theano/sandbox/multinomial.py

```diff
@@ -45,18 +47,18 @@ def make_node(self, pvals, unis):
         else:
             odtype = self.odtype
         out = T.tensor(dtype=odtype, broadcastable=pvals.type.broadcastable)
-        return Apply(self, [pvals, unis], [out])
+        return Apply(self, [pvals, unis, as_scalar(n)], [out])
 
     def grad(self, ins, outgrads):
-        pvals, unis = ins
+        pvals, unis, n = ins
         (gz,) = outgrads
         return [T.zeros_like(x) for x in ins]
 
     def c_code_cache_version(self):
-        return (5,)
+        return (6,)
 
     def c_code(self, node, name, ins, outs, sub):
-        (pvals, unis) = ins
+        (pvals, unis, n) = ins
```

Comment:
As n is a scalar.Scalar and not a tensor.scalar, to support old pickled graph, doing the same as in perform() should work. Just test it.